state/application.go

```diff
@@ -2527,17 +2544,29 @@ func (op *AddUnitOperation) Done(err error) error {
 			return errors.Trace(err)
 		}
 	}
-	if op.props.UnitStatus != nil {
-		now := op.application.st.clock().Now()
-		if err := u.SetStatus(status.StatusInfo{
-			Status:  op.props.UnitStatus.Status,
-			Message: op.props.UnitStatus.Message,
-			Data:    op.props.UnitStatus.Data,
-			Since:   &now,
-		}); err != nil {
+	if op.props.CloudContainerStatus != nil {
```

Comment:
It seems we're failing to update container status history

Reply:
Ah good catch as the createStatusOps doesn't handle the history.

Reply:
Yep, and the tests don't check for it either. They only load and test unit status history